engine/api/src/main/scala/pl/touk/nussknacker/engine/api/context/transformation/DefinedParameter.scala

```diff
@@ -1,14 +1,29 @@
 package pl.touk.nussknacker.engine.api.context.transformation
 
-import pl.touk.nussknacker.engine.api.typed.typing.{TypingResult, Unknown}
+import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypingResult, Unknown}
 
 sealed trait DefinedParameter {
   def returnType: TypingResult
 }
 
-case class DefinedLazyParameter(returnType: TypingResult) extends DefinedParameter
-case class DefinedEagerParameter(value: Any, returnType: TypingResult) extends DefinedParameter
-case object FailedToDefineParameter extends DefinedParameter {
-  override def returnType: TypingResult = Unknown
+sealed trait DefinedSingleParameter extends DefinedParameter
+
+sealed trait DefinedBranchParameter extends DefinedParameter {
```

Comment:
Well, I changed it. But it adds not so much value, because we cannot statically check if transformation returns branch or not - in parameter definition type is controlled by flag. We would have to change api.definition.Parameter to class hierarchy, and this would be quite large change...

Reply:
We are one step closer to be safe - it is fair enough for me :)